internal/builder/external_fields.go

```diff
@@ -37,11 +37,18 @@ func resolveExternalFields(packageRoot, destinationDir string) error {
 		return errors.Wrap(err, "can't create field dependency manager")
 	}
 
-	fieldsFile, err := filepath.Glob(filepath.Join(destinationDir, "data_stream", "*", "fields", "*.yml"))
+	dataStreamFieldsFiles, err := filepath.Glob(filepath.Join(destinationDir, "data_stream", "*", "fields", "*.yml"))
 	if err != nil {
 		return err
 	}
-	for _, file := range fieldsFile {
+
+	packageFieldsFiles, err := filepath.Glob(filepath.Join(destinationDir, "fields", "*.yml"))
```

Comment:
Maybe we should refactor `resolveExternalFields` to accept only the paths where fields are, this way the method can be reused for any set of directories containing fields files.

Then the method to build integration packages would call `resolveExternalFields` with the data stream directories, and the method to build input packages would call it with its fields directory.

I see we are reusing the same `BuildPackage` method for both types of packages, maybe we should have different build methods for each kind of package.

This can be left for a future refactor, maybe when we have a third kind of package with fields :)

Reply:
It sounds like a good idea, I can create an issue for this as I don't want to swamp in that refactoring while introducing input packages.

Reply:
https://github.com/elastic/elastic-package/issues/843